x/rollapp/abci.go

```diff
@@ -17,41 +17,57 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper)
 
 // Called every block to finalize states that their dispute period over.
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
+
 	// check to see if there are pending  states to be finalized
-	blockHeightToFinalizationQueue, found := k.GetBlockHeightToFinalizationQueue(ctx, uint64(ctx.BlockHeight()))
-	if !found {
-		return
+	//blockHeightToFinalizationQueue, found := k.GetBlockHeightToFinalizationQueue(ctx, uint64(ctx.BlockHeight()))
+	globalIndex, found := k.GetLatestFinalizedStateGlobalIndex(ctx)
+
+	finalizationHeight := uint64(0)
+	if uint64(ctx.BlockHeight()) >= k.DisputePeriodInBlocks(ctx) {
+		finalizationHeight = uint64(ctx.BlockHeight()) - k.DisputePeriodInBlocks(ctx)
 	}
 
-	// finalize pending states
-	for _, stateInfoIndex := range blockHeightToFinalizationQueue.FinalizationQueue {
-		stateInfo, found := k.GetStateInfo(ctx, stateInfoIndex.RollappId, stateInfoIndex.Index)
+	for found && globalIndex.Index <= finalizationHeight {
```

Comment:
don't understand this. what is the goal of global index and why is it compared to finalizationHeight?

Reply:
outdated code